### ISSUE TITLE
[Feature Request]: Alert Features missing - Clearing/Silencing/Saving

### ISSUE THREAD

**Issue by varaga** [grafana/grafana#9106]
Read before posting: 

- Questions should be posted to https://community.grafana.com. Please search there and here on GitHub for similar issues before creating a new issue. 
- Checkout FAQ: https://community.grafana.com/c/howto/faq
- Checkout How to troubleshoot metric query issues: https://community.grafana.com/t/how-to-troubleshoot-metric-query-issues/50

Please prefix your title with [Bug] or [Feature request]. 

Please include this information:
- What Grafana version are you using? 4.1.1
- What datasource are you using? Prometheus / InfluxDb
- What OS are you running grafana on? RH Linux
- What did you do? Set up Alerts based on metrics sourced through influx db and Prometheus
- What was the expected result? Missing features as outlined below
- What happened instead?
- If related to metric query / data viz:
  - Include raw network request & response: get by opening Chrome Dev Tools (F12, Ctrl+Shift+I on windows, Cmd+Opt+I on Mac), go the network tab.

I'm evaluating use of Alerts through Grafana as apposed to using other tools.
Basically, I have set up grafana to pull metrics from Prometheus and have set up alerts (integrated with slack for notifications). 
The Query is a rate calculation that is set to evaluate every 30s.
Alert is set for avg() of query(A, 5m, now). I understand that this runs every 30s and evaluates the overlap from the previous metric. 

 1. Is there anyway to apply a condition or a count over a sliding window.. for eg., I do not want to alert for a once-off spike, however, I want to raise an alarm when it happens 3 times over 30 minutes?

 2. Is there a way to silence the alert for a time range?

 3. Is there a way to clear the state (automated house-cleaning)

 4. Is there a way to save the alerts into a data store?

**Comment by torkelo** [comment]
> Is there anyway to apply a condition or a count over a sliding window.. for eg., I do not want to alert for a once-off spike, however, I want to raise an alarm when it happens 3 times over 30 minutes?

You can get close to this by using median, or have one query for the metric and one for the standard deviation. But it' stricky to get the same behavior as just "only alert if the rule evaluates to true 3 times in a row", there is a feature request for that: 
https://github.com/grafana/grafana/issues/7320

> Is there a way to silence the alert for a time range?
Not yet, but we want to implement it,  https://github.com/grafana/grafana/issues/5856

> Is there a way to clear the state (automated house-cleaning)

Not a global way. There is an api to delete state history per dashboard & panel (used by UI in the history view of the alert tab) a but it's not documented yet .

**Comment by torkelo** [comment]
> Is there a way to save the alerts into a data store?

You mean the notifications? Not out of the box. You can use the built in webhook notification and build a small app that receives those web requests and stores them in your db of choice
